salt/states/boto_asg.py

```diff
@@ -851,41 +870,86 @@ def absent(
+        )
+    else:
+        changes = {'old': name, 'new': None}
         if __opts__['test']:
-            ret['comment'] = 'Autoscale group set to be deleted.'
             ret['result'] = None
-            if remove_lc:
-                msg = 'Launch configuration {0} is set to be deleted.'.format(asg['launch_config_name'])
-                ret['comment'] = ' '.join([ret['comment'], msg])
-            return ret
-        deleted = __salt__['boto_asg.delete'](name, force, region, key, keyid,
-                                              profile)
-        if deleted:
-            if remove_lc:
-                lc_deleted = __salt__['boto_asg.delete_launch_configuration'](asg['launch_config_name'],
-                                                                              region,
-                                                                              key,
-                                                                              keyid,
-                                                                              profile)
-                if lc_deleted:
-                    if 'launch_config' not in ret['changes']:
-                        ret['changes']['launch_config'] = {}
-                    ret['changes']['launch_config']['deleted'] = asg['launch_config_name']
-                else:
-                    ret['result'] = False
-                    ret['comment'] = ' '.join([ret['comment'], 'Failed to delete launch configuration.'])
-            ret['changes']['old'] = asg
-            ret['changes']['new'] = None
-            ret['comment'] = 'Deleted autoscale group.'
+            ret['comment'].append(
+                'Autoscale group {0} set to be deleted.'.format(name)
+            )
+            ret['pchanges'] = {'asg': changes}
         else:
-            ret['result'] = False
-            ret['comment'] = 'Failed to delete autoscale group.'
-    else:
-        ret['comment'] = 'Autoscale group does not exist.'
+            deleted = __salt__['boto_asg.delete'](
+                name,
+                force=force,
+                region=region,
+                key=key,
+                keyid=keyid,
+                profile=profile,
+            )
+            if not deleted:
+                ret['result'] = False
+                ret['comment'].append(
+                    'Failed to delete autoscale group {0}.'.format(name)
+                )
+                return ret
+
+            ret['result'] = True
+            ret['comment'].append('Deleted autoscale group {0}.'.format(name))
+            ret['changes']['asg'] = changes
+
+    if remove_lc or recurse:
+        _ret = __states__['boto_lc.absent'](
+            asg['launch_config_name'],
+            region=region,
+            key=key,
+            keyid=keyid,
+            profile=profile,
+        )
+        __utils__['state.merge_subreturn'](ret, _ret, subkey='launch_config')
+        if ret['result'] is False:
```

Comment:
This should be `if _ret['result'] is False:` here, right? I don't see where `ret['result']` could be `False` at this point because it should have already returned.